vscode-test-runner/workspace-1/.joyride/src/integration_test/index.cljs

```diff
@@ -0,0 +1,52 @@
+(ns integration-test.index
+  (:require [cljs.test]
+            [promesa.core :as p]))
+
+(def !results (atom {:fail 0
+                     :error 0}))
+
+(defmethod cljs.test/report [:cljs.test/default :begin-test-var] [m]
+  (println "===" (-> m :var meta :name))
+  (println))
+
+(def old-fail (get-method cljs.test/report [:cljs.test/default :fail]))
+
+(defmethod cljs.test/report [:cljs.test/default :fail] [m]
+  (old-fail m)
+  (swap! !results update :fail inc))
+
+(def old-error (get-method cljs.test/report [:cljs.test/default :fail]))
+
+(defmethod cljs.test/report [:cljs.test/default :error] [m]
+  (old-error m)
+  (swap! !results update :error inc))
+
+(defn run-all-tests []
+  (p/create (fn [resolve, reject]
+              (p/do!
+               (p/delay 1000) ; This waits for workspace_activate.cljs to run
```

Comment:
Using a timeout seems a recipe for flaky tests. Is there a more reliable way?

Reply:
I don't know what that would be. Maybe some side effect we can make happen from the activation script...

Reply:
I pushed a change where our test runner creates a promise that the workspace-activate script resolves, and then the tests are run. If this promise is not created when the workspace-activate script has run, it will try again every 500 ms. I've not seen this happen, but I don't think we can rely on it always being like that.